concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/window/PercentileSampleWindowFactory.java

```diff
@@ -19,18 +19,20 @@
 
 public class PercentileSampleWindowFactory implements SampleWindowFactory {
     private final double percentile;
+    private final int windowSize;
 
-    private PercentileSampleWindowFactory(double percentile) {
+    private PercentileSampleWindowFactory(double percentile, int windowSize) {
         this.percentile = percentile;
+        this.windowSize = windowSize;
     }
 
-    public static PercentileSampleWindowFactory of(double percentile) {
+    public static PercentileSampleWindowFactory of(double percentile, int windowSize) {
```

Comment:
I am struggling a lot about how to design this part of api, I left it in a way, so instantiation would look like this:
```
WindowedLimit limit = new WindowedLimit.Builder()
                .sampleWindowFactory(PercentileSampleWindowFactory.of(0.9, 50))
                .windowSize(50)
                .build(AIMDLimit.newBuilder().build())
```
because I am not sure which one is better(no overengineering, protect from user errors) amongst possible solutions:
1. Keep it as is. Duplicate code for end-user, error-prone
2. pass windowSize parameter to `newInstance()` call, it seems like factory should be aware about this value during creation process and bring no burden to WindowedLimit code
3. introduce `setWindowSize` method for SampleWindowFactory. Messy, since factory object is not immutable anymore
4. introduce SampleWindowFactoryBuilder. Might looks good unless people would see, that they should pass Builder and bother about how to create it (hide it behind SampleWindowFactoryBuilderFactory probably), but this one seems too complex

@elandau Maybe I am missing something better or making mistakes while comparing those solutions?

Reply:
I think it's fine the way it is.  I feel that anything beyond this would just complicate the API for little added benefit.